src/pyodide/python-entrypoint-helper.js

```diff
@@ -54,49 +54,63 @@ function initializePackageIndex(pyodide) {
 
 // These packages are currently embedded inside EW and so don't need to
 // be separately installed.
-const EMBEDDED_PYTHON_PACKAGES = new Set([
-  "aiohttp",
-  "aiosignal",
-  "anyio",
-  "async_timeout",
-  "attrs",
-  "certifi",
-  "charset_normalizer",
-  "dataclasses_json",
-  "distro",
-  "fastapi",
-  "frozenlist",
-  "h11",
-  "httpcore",
-  "httpx",
-  "idna",
-  "jsonpatch",
-  "jsonpointer",
-  "langchain",
-  "langchain_community",
-  "langchain_core",
-  "langsmith",
-  "marshmallow",
-  "micropip",
-  "multidict",
-  "mypy_extensions",
-  "numpy",
-  "openai",
-  "packaging",
-  "pydantic",
-  "PyYAML",
-  "requests",
-  "setuptools",
-  "sniffio",
-  "SQLAlchemy",
-  "starlette",
-  "tenacity",
-  "tqdm",
-  "typing_extensions",
-  "typing_inspect",
-  "urllib3",
-  "yarl",
-]);
+// If a version number is specified (i.e. non-empty), it is passed to micropip
+// when running in local dev.
+const EMBEDDED_PYTHON_PACKAGES = {
+  "aiohttp": "3.9.3",
+  "aiosignal": "1.3.1",
+  "anyio": "3.7.1",
+  "attrs": "23.2.0",
+  "certifi": "2024.2.2",
+  "charset_normalizer": "3.3.2",
```

Comment:
@dom96 @hoodmane 

I kept the underscores since that's what the original code said, but I think it might be necessary to [normalize package names](https://packaging.python.org/en/latest/specifications/name-normalization/) in this list as well as during lookup. But that might be better to put in another PR.

Reply:
We probably want to have pretty similar logic to `recursiveDependencies` here:
https://github.com/pyodide/pyodide/blob/main/src/js/load-package.ts?plain=1#L186

In fact there's a pretty good chance we could use that same logic unchanged.

Reply:
Good point, yeah, normalization is a must.